File: design-patterns/csharp/templatemethod/ReportData.cs
using System.Collections.Generic;

namespace TemplateMethod
{
    public class ReportData
    {
        public List<string> GetHeaders()
        {
            return new List<string> { "ID", "Name", "Value" };
        }

        public List<Dictionary<string, object>> GetRows()
        {
            return new List<Dictionary<string, object>>
            {
                new Dictionary<string, object> { { "ID", 1 }, { "Name", "Item A" }, { "Value", 100.0 } },
                new Dictionary<string, object> { { "ID", 2 }, { "Name", "Item B" }, { "Value", 150.5 } },
                new Dictionary<string, object> { { "ID", 3 }, { "Name", "Item C" }, { "Value", 75.25 } }
            };
        }
    }
} 

File: design-patterns/cpp/templatemethod/report_data.cpp
#include "report_data.h"

std::vector<std::string> ReportData::getHeaders() const {
    return {"ID", "Name", "Value"};
}

std::vector<std::map<std::string, std::variant<int, double, std::string>>> ReportData::getRows() const {
    return {
        {
            {"ID", 1},
            {"Name", "Item A"},
            {"Value", 100.0}
        },
        {
            {"ID", 2},
            {"Name", "Item B"},
            {"Value", 150.5}
        },
        {
            {"ID", 3},
            {"Name", "Item C"},
            {"Value", 75.25}
        }
    };
} 